app/src/main/java/mozilla/lockbox/presenter/ItemDetailPresenter.kt

```diff
@@ -11,77 +11,79 @@ import io.reactivex.Observable
-                .subscribe {
-                    view.itemId?.let {
-                        dataStore.get(it)
-                                .subscribe {
-                                    if (it!!.username!!.isNotEmpty()) {
-                                        dispatcher.dispatch(ClipboardAction.CopyUsername(it.username!!))
-                                        view.showToastNotification(R.string.toast_username_copied)
-                                    }
-                                }
-                                .addTo(compositeDisposable)
-                    }
-                }
-                .addTo(compositeDisposable)
+        handleClicks(view.usernameCopyClicks) {
+            val username = it.username ?: ""
+            if (username.isNotEmpty()) {
+                dispatcher.dispatch(ClipboardAction.CopyUsername(username))
+                view.showToastNotification(R.string.toast_username_copied)
+            }
+        }
 
-        this.view.passwordCopyClicks
-                .subscribe {
-                    view.itemId?.let {
-                        dataStore.get(it)
-                                .subscribe {
-                                    if (it!!.password.isNotEmpty()) {
-                                        dispatcher.dispatch(ClipboardAction.CopyPassword(it.password))
-                                        view.showToastNotification(R.string.toast_password_copied)
-                                    }
-                                }
-                                .addTo(compositeDisposable)
-                    }
-                }
-                .addTo(compositeDisposable)
+        handleClicks(view.passwordCopyClicks) {
+            if (it.password.isNotEmpty()) {
+                dispatcher.dispatch(ClipboardAction.CopyPassword(it.password))
+                view.showToastNotification(R.string.toast_password_copied)
+            }
+        }
+
+        handleClicks(view.hostnameClicks) {
+            if (it.hostname.isNotEmpty()) {
+                dispatcher.dispatch(RouteAction.OpenWebsite(it.hostname))
+            }
+        }
 
         this.view.togglePasswordClicks
-                .subscribe {
-                    view.isPasswordVisible = view.isPasswordVisible.not()
-                }
-                .addTo(compositeDisposable)
+            .subscribe {
+                view.isPasswordVisible = view.isPasswordVisible.not()
+            }
+            .addTo(compositeDisposable)
+
+        view.isPasswordVisible = false
 
         // now set up the data.
-        val itemId = view.itemId ?: return
+        val itemId = this.itemId ?: return
         dataStore.get(itemId)
-                .map {
-                    ItemDetailViewModel(it.id, titleFromHostname(it.hostname), it.hostname, it.username, it.password)
-                }
-                .subscribe(view::updateItem)
-                .addTo(compositeDisposable)
+            .map {
```

Comment:
Also, we should make sure to bind this sequence to the main thread so that we don't accidentally update the view from a background thread.